_source/_posts/2022-06-16-oauth-java.adoc

```diff
@@ -0,0 +1,389 @@
+
+=== Spring Boot
+
+Spring Boot uses Spring Security by default and it's the only framework that allows you to configure a resource server with one line of code.
+
+[source,properties]
+----
+spring.security.oauth2.resourceserver.jwt.issuer-uri=https://dev-13337.okta.com/oauth2/default
+----
+
+Of course, you could use YAML instead. That would be a bad life choice. Then it would require six lines of code. Don't be that person. Spring Security is smart enough to look up the JWKS location from the issuer.
+
+To be fair, Spring Security does require some Java code to complete the configuration, while the others do everything automatically.
+
+[source,java]
+----
+@EnableWebSecurity
+public class SecurityConfiguration {
+
+    @Bean
+    public SecurityWebFilterChain springSecurityFilterChain(ServerHttpSecurity http) {
+        http
+            .authorizeRequests(request -> request.anyRequest().authenticated())
+            .oauth2ResourceServer(OAuth2ResourceServerConfigurer::jwt);
+        return http.build();
+    }
+}
+----
+
+TIP: If you use the https://github.com/okta/okta-spring-boot[Okta Spring Boot Starter], you do not need to configure things with Java. We've included this configuration for you.
+
+From this information, I hope you have a decent overview of where Java is when it comes to OAuth. If you like to learn by watching, reading, or doing&mdash;keep reading!
+
+== Watch OAuth 2.0 in action with Java frameworks
+
+I've created a few screencasts to show how to build a REST API with Quarkus, Spring Boot, Micronaut, and Helidon. These videos also show how to secure each API with OAuth 2.0.
+
+++++
+<div class="video-grid">
+  <div class="video">
+    <iframe width="267" height="150" src="https://www.youtube.com/embed/3J9uFVd3BoY" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+  </div>
+  <div class="video">
+    <iframe width="267" height="150" src="https://www.youtube.com/embed/w-qKailh3WQ" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+  </div>
+  <div class="video">
+    <iframe width="267" height="150" src="https://www.youtube.com/embed/IG2uo4IP1QI" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+  </div>
+  <div class="video">
+    <iframe width="267" height="150" src="https://www.youtube.com/embed/Bj69rOIwQwY" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+  </div>
+</div>
+++++
+
+At the end of each video, I show how the https://cli.okta.com[Okta CLI] allows you to create working examples for each framework with the `okta start` command.
+
+If you prefer reading and copy pasta over video, check out these quick tutorials:
```

Comment:
Can you italicize "copy pasta" - maybe it's just me, but I always think it's a typo at first when I see it